### PR TITLE
Protocol sniffer arp tcp

### PR DESCRIPTION
**Pull request by PixiesRAC**
Adding UDP/TCP/ARP PROTOCOL AND FILTER

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -41,6 +41,17 @@ set(INCLUDE_JSONBOOST_DIR, third_party/property_tree/include/)
 # END Json Boost
 ################################
 
+option(WITH_FILTER "Filter" 1) #FILTER SET VALUE TO 0 TO UNABLE THE FILTER
```
Comment (PixiesRAC) [inline CMakeLists.txt:44]: disable

File: include/filter/FilterOutput.h [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+{
+    class FilterOutput
+    {
+	private :
+
+	FilterOutput();
+
+	public :
+
+	~FilterOutput();
```
Comment (vguen) [inline include/filter/FilterOutput.h:17]: So anyone can call your singleton destructor ? It should a private member

File: include/filter/FilterOutput.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		   return data;
+	       }
+	       else
+	       {
+		   return "";
+	       }
+	}
+	
+	static std::vector<std::string> filterKeywords;
+	static FilterOutput		*pInstance;
```
Comment (vguen) [inline include/filter/FilterOutput.h:43]: Your instance is public static : anyone can use it, destroy it.. (should be private)

File: include/protocols/ProtocolTCP.h [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	    ProtocolTCP();
+	    ~ProtocolTCP();
+
+	    using protocol = struct tcphdr;
+
+	private :
+	
+	    friend class IProtocol<ProtocolTCP>;
+
+	    size_t	derivedGetStructSize() const
```
Comment (vguen) [inline include/protocols/ProtocolTCP.h:21]: Implementation should be in cpp file

File: include/protocols/ProtocolUDP.h [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+	ProtocolUDP();
+	~ProtocolUDP();
+
+	using protocol = struct udphdr;
+
+	private :
+
+	friend class IProtocol<ProtocolUDP>;
+
+	size_t	derivedGetStructSize() const // modifier le nom avec derived
```
Comment (vguen) [inline include/protocols/ProtocolUDP.h:22]: Implementation should be in cpp file

File: include/filter/FilterOutput.h (hunk 2 above)
Comment (PixiesRAC) [reply include/filter/FilterOutput.h:17]: i'm using an unique_ptr for this class, it's for that i let the destructor public

**Pull request merged**